modemmanger: compilation error

Maintainer: @nickberry17 
Environment: x86_64, APU3, current OpenWrt master

Description:
I wanted to evaluate the modemmanger to replace my current setup uqmi proto with the modemmanager. On my development machine (debian 9) compilation works.
But On my compilation server machine (also debian 9) with minimal setup the compilation of modemmanger failes. I think that hast todo with the modemmanager i am using on my development machine for network connectivity so that the compilation is not independent. On the compilation server no networkmanager ist installed.

Error output of the compilation:

```

/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/staging_dir/target-x86_64_musl/usr/bin/glib-mkenums \
        --fhead "#include <ModemManager.h>\n#ifndef __MM_ENUMS_TYPES_H__\n#define __MM_ENUMS_TYPES_H__\n" \
        --template ../../build-aux/mm-enums-template.h \
        --ftail "#endif /* __MM_ENUMS_TYPES_H__ */\n" \
        ../../include/ModemManager-enums.h > mm-enums-types.h
/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/staging_dir/target-x86_64_musl/usr/bin/glib-mkenums \
        --fhead "#ifndef __MM_ERRORS_TYPES_H__\n#define __MM_ERRORS_TYPES_H__\n" \
        --template ../../build-aux/mm-errors-template.h \
        --ftail "#endif /* __MM_ERRORS_TYPES_H__ */\n" \
        ../../include/ModemManager-errors.h > mm-errors-types.h
/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/staging_dir/target-x86_64_musl/usr/bin/gdbus-codegen \
        --interface-prefix org.freedesktop.ModemManager1. \
        --c-namespace=MmGdbus \
        --generate-docbook mm-gdbus-doc \
        --generate-c-code mm-gdbus-manager \
        ../../introspection/org.freedesktop.ModemManager1.xml \

Traceback (most recent call last):
  File "/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/staging_dir/target-x86_64_musl/usr/bin/gdbus-codegen", line 53, in <module>
    from codegen import codegen_main
ImportError: No module named codegen
Makefile:1149: recipe for target 'mm-gdbus-manager.c' failed
make[6]: *** [mm-gdbus-manager.c] Error 1
make[6]: Leaving directory '/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/build_dir/target-x86_64_musl/ModemManager-1.10.6/libmm-glib/generated'
Makefile:1250: recipe for target 'all-recursive' failed
make[5]: *** [all-recursive] Error 1
make[5]: Leaving directory '/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/build_dir/target-x86_64_musl/ModemManager-1.10.6/libmm-glib'
Makefile:510: recipe for target 'all-recursive' failed
make[4]: *** [all-recursive] Error 1
make[4]: Leaving directory '/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/build_dir/target-x86_64_musl/ModemManager-1.10.6'
Makefile:442: recipe for target 'all' failed
make[3]: *** [all] Error 2
make[3]: Leaving directory '/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/build_dir/target-x86_64_musl/ModemManager-1.10.6'
Makefile:130: recipe for target '/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/build_dir/target-x86_64_musl/ModemManager-1.10.6/.built' failed
make[2]: *** [/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/build_dir/target-x86_64_musl/ModemManager-1.10.6/.built] Error 2
make[2]: Leaving directory '/home/bbworker/bbworker/owrt_master_x86_64/System6/build/feeds/packages/net/modemmanager'
time: package/feeds/packages/modemmanager/compile#4.80#2.28#10.70
package/Makefile:111: recipe for target 'package/feeds/packages/modemmanager/compile' failed
make[1]: *** [package/feeds/packages/modemmanager/compile] Error 2
make[1]: Leaving directory '/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt'
/home/bbworker/bbworker/owrt_master_x86_64/System6/build/openwrt/include/toplevel.mk:225: recipe for target 'package/feeds/packages/modemmanager/compile' failed
make: *** [package/feeds/packages/modemmanager/compile] Error 2

```

--- a/net/modemmanager/Makefile
+++ b/net/modemmanager/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=modemmanager
 PKG_VERSION:=1.10.6
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE:=ModemManager-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=https://www.freedesktop.org/software/ModemManager
@@ -22,6 +22,10 @@ LICENSE_FILES:=COPYING
 
 PKG_INSTALL:=1
 PKG_BUILD_PARALLEL:=1
+
+PKG_BUILD_DEPENDS:=glib2/host libxslt/host
+
+PKG_FIXUP:=autoreconf
 
 include $(INCLUDE_DIR)/package.mk
 
@@ -73,6 +77,10 @@ define Build/Prepare
 	( cd "$(PKG_BUILD_DIR)"; \
 		printf "all:\ninstall:\n" >po/Makefile.in.in; \
 	)
+	$(SED) 's|^\(GLIB_MKENUMS\)=.*|\1=$(STAGING_DIR_HOSTPKG)/bin/glib-mkenums|' \
+		$(PKG_BUILD_DIR)/configure.ac
+	$(SED) 's|^\(GDBUS_CODEGEN\)=.*|\1=$(STAGING_DIR_HOSTPKG)/bin/gdbus-codegen|' \
+		$(PKG_BUILD_DIR)/configure.ac
 endef
 
 define Build/InstallDev
@@ -124,9 +132,5 @@ define Package/modemmanager/install
 	$(INSTALL_DIR) $(1)/lib/netifd/proto
 	$(INSTALL_BIN) ./files/modemmanager.proto $(1)/lib/netifd/proto/modemmanager.sh
 endef
-
-$(eval $(call RequireCommand,xsltproc, \
-        $(PKG_NAME) requires xsltproc installed on the host-system. \
-))
 
 $(eval $(call BuildPackage,modemmanager))

